app/main/views/reset_password.py

```diff
@@ -213,7 +214,16 @@ def change_password():
 
                 form.old_password.errors.append("Make sure you’ve entered the right password.")
 
-        return render_template("auth/change-password.html", form=form, dashboard_url=dashboard_url), 400
+        if form.errors:
+            # Error messages should have the same order as the fields (specified by 'id' prefix)
+            form_error_fields = sorted([field_name for field_name in form.errors.keys()], key=lambda x: form[x].id)
+
+        return render_template(
+            "auth/change-password.html",
+            form=form,
+            form_error_fields=form_error_fields,
```

Comment:
What format are we expecting `errors` to be? Here's it's just a list of field names - that might be sufficient if we're passing the form through as well and the toolkit can access the errors themselves (multiple errors per field and so on). Is there an example somewhere of where we use `errors` for a WTForm (as opposed to a content loader form)?

Reply:
I'm not overly concerned about the format of `errors`, as long as it will evaluate to boolean True/False, which should let us work out if the page is in an 'error state' or not.

Reply:
Looking at the WTForms in the Briefs award flow, we use a nested dict for fields with potentially multiple errors:

```
errors = {
    key: {'question': form[key].label.text, 'input_name': key, 'message': form[key].errors[0]}
    for key, value in form.errors.items()
}
```

This then feeds into the toolkit validation template (as we eventually hope to do here). I'll try changing my list of strings into an OrderedDict so we can maintain our special ordering and re-use the error messages properly later.